tensornetwork/contractors/opt_einsum_paths/path_contractors_node_test.py

```diff
@@ -96,3 +96,28 @@ def __call__(self, inputs, output, size_dict, memory_limit=None):
   optimizer = PathOptimizer()
   final_node = path_contractors.custom(nodes, optimizer)
   np.testing.assert_allclose(final_node.tensor, np.ones(5) * 2.0)
+
+
+def test_subgraph_contraction(backend, path_algorithm):
+  a = tn.Node(np.ones((2, 2)), backend=backend)
+  b = tn.Node(np.ones((2, 2)), backend=backend)
+  c = tn.Node(np.ones((2, 2)), backend=backend)
+  a[0] ^ b[0]
+  c[0] ^ b[1]
+  result = path_algorithm({a, b}, [a[1], b[1]])
```

Comment:
Could you add an assert verifying that this intermediate result is correct?

Reply:
Added